### PR TITLE
SCE: X-Forwarded-For takes priority over Forwarded: for= and remove port

### PR DESCRIPTION
**Pull request by benjben**

### CODE REVIEW

File: 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/loaders/IpAddressExtractor.scala [hunk 1]
```diff
@@ -30,20 +31,29 @@ object IpAddressExtractor {
...
     headers match {
       case h :: t =>
         h.toLowerCase match {
-          case XForwardedForRegex(originalIpAddress) => originalIpAddress
-          case ForwardedForRegex(originalIpAddress)  => originalIpAddress
-          case _                                     => extractIpAddress(t, lastIp)
+          case XForwardedForRegex(ipv4, ipv6) => Option(ipv4).getOrElse(ipv6)
+          case ForwardedForRegex(ipv4, ipv6) =>
+            val ip = Option(ipv4).getOrElse(ipv6)
+            extractIpAddress(t, lastIp, Some(ip))
```
Comment (BenFradet) [inline 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/loaders/IpAddressExtractor.scala]: this should be `Option(ip)` no? can the two groups be null?

File: 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/loaders/IpAddressExtractor.scala [hunk 2]
```diff
@@ -30,20 +31,29 @@ object IpAddressExtractor {
...
   @tailrec
-  def extractIpAddress(headers: List[String], lastIp: String): String =
+  def extractIpAddress(headers: List[String], lastIp: String, maybeForwardedForIp: Option[String] = None): String =
     headers match {
       case h :: t =>
         h.toLowerCase match {
-          case XForwardedForRegex(originalIpAddress) => originalIpAddress
-          case ForwardedForRegex(originalIpAddress)  => originalIpAddress
-          case _                                     => extractIpAddress(t, lastIp)
+          case XForwardedForRegex(ipv4, ipv6) => Option(ipv4).getOrElse(ipv6)
```
Comment (BenFradet) [inline 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/loaders/IpAddressExtractor.scala]: same, can the two groups be null?

File: 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/loaders/IpAddressExtractor.scala (hunk 1 above)
Comment (benjben) [reply 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/loaders/IpAddressExtractor.scala]: If the 2 groups were null the regex would not pattern match.
```
scala> val Regex = """(a{2})|(a{3})""".r
Regex: scala.util.matching.Regex = (a{2})|(a{3})

scala> "aa" match { case Regex(group1, group2) => (group1, group2); case _ => "no match" }
res1: java.io.Serializable = (aa,null)

scala> "aaa" match { case Regex(group1, group2) => (group1, group2); case _ => "no match" }
res2: java.io.Serializable = (null,aaa)

scala> "aaaa" match { case Regex(group1, group2) => (group1, group2); case _ => "no match" }
res3: java.io.Serializable = no match
```

**Comment by benjben** [comment]
Superseded by #4076

**Pull request closed without merging**